modules/protocol/src/vector.ts

```diff
@@ -332,6 +393,97 @@ export class Vector implements IVectorProtocol {
     return undefined;
   }
 
+  // Adds a given task to the internal queue
+  // TODO: implement
+  private addToQueue(
+    params: UpdateParams<any>,
+    previous?: ChannelUpdate,
+  ): Promise<
+    Result<
+      {
+        updatedChannel: FullChannelState;
+        updatedActiveTransfers?: FullTransferState[];
+        updatedTransfer?: FullTransferState;
+      },
+      QueuedUpdateError
+    >
+  > {
+    throw new Error("addToQueue method not implemented");
+  }
+
+  // Adds a given task to the front of the internal queue
+  // TODO: implement
+  private addToFrontOfQueue(
+    params: UpdateParams<any>,
+    previous?: ChannelUpdate,
+  ): Promise<
+    Result<
+      {
+        updatedChannel: FullChannelState;
+        updatedActiveTransfers?: FullTransferState[];
+        updatedTransfer?: FullTransferState;
+      },
+      QueuedUpdateError
+    >
+  > {
+    throw new Error("addToQueue method not implemented");
+  }
+
+  /**
+   * Returns true if the received upddate includes our proposed update. If true,
+   * this means you don't have to re-send your proposed update. If false, this
+   * means you should resend your proposed upddate
+   * @param receivedUpdate The update you have gotten from your counterparty
+   * @param proposedUpdate The update you have sent out to counterparty
+   */
+  private includesOurProposedUpdate(
+    receivedUpdate: ChannelUpdate,
+    proposedUpdate: ChannelUpdate,
+    // TODO: should probably just use the generated merkle tree here instead
+    // of regenerating it from the transfers and channel
+    updatedActiveTransfers: FullTransferState[],
+    proposedTransfer?: FullTransferState,
+  ): Result<boolean, QueuedUpdateError> {
+    // If both are a setup update, your update would fail and the ultimate
+    // result (a channel is set up) is achieved
+    if (receivedUpdate.type === UpdateType.setup && proposedUpdate.type === UpdateType.setup) {
+      return Result.ok(true);
+    }
+
+    // If both are a deposit update, your deposit is implicitly included
+    if (receivedUpdate.type === UpdateType.deposit && proposedUpdate.type === UpdateType.deposit) {
+      return Result.ok(true);
+    }
+
+    // If both are a resolve, it would *not* include your proposed update since
+    // only the responder to a transfer can call resolve. Updates by definition
+    // are different
+    if (receivedUpdate.type === UpdateType.resolve && proposedUpdate.type === UpdateType.resolve) {
+      return Result.ok(false);
+    }
+
+    // If both are a create, it would include your proposed update IFF the
+    // merkle root *after* the update was applied includes the transfer you
+    // attempted to create
+    if (receivedUpdate.type === UpdateType.create && proposedUpdate.type === UpdateType.create) {
```

Comment:
The first check to see if both are setup makes sense, but all these branches after I'm a bit lost.

The way I see it there's 3 possibilities and they are independent of the types of updates.
  1. The final state is (initial + our update + their update) => return true
  2. The final state is (initial + their update) => return false
  3.  The final state is something else => fail
This can be detected by just playing the updates forward and seeing what the merkle root is.